drivers/serial/uart_ifx_cat1.c

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief UART driver for Infineon CAT1 MCU family.
+ *
+ * Note:
+ * - Uart ASYNC functionality is not implemented in current
+ *   version of Uart CAT1 driver.
+ */
+
+#define DT_DRV_COMPAT infineon_cat1_uart
+
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <cyhal_uart.h>
+#include <cyhal_utils_psoc.h>
+#include <cyhal_scb_common.h>
+
+/* Default UART interrupt priority */
+#if !defined(IFX_CAT1_UART_INTERRUPT_PRIORITY)
+	#define IFX_CAT1_UART_INTERRUPT_PRIORITY      (5u)
+#endif
+
+/* Data structure */
+struct ifx_cat1_uart_data {
+	cyhal_uart_t obj;                              /* UART CYHAL object */
+	struct uart_config cfg;
+	cyhal_clock_t clock;
+
+	#if CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_cb;           /* Interrupt Callback */
+	void *irq_cb_data;                              /* Interrupt Callback Arg */
+	#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
```

Comment:
`#if` usually starts at col 0 (or check with clang-format results)

Reply:
Updated. I am using uncrustify, but now i see Zephyr does not provide uncrustify configuration... I will look and integrate in our CI clang-format.